packages/legend-extension-dsl-service/src/components/query/ServiceRegisterModal.tsx

```diff
@@ -109,56 +115,49 @@ const ServiceRegisterModal = observer(
 
     const registerService = editorStore.applicationStore.guardUnhandledError(
       async (): Promise<void> => {
-        const project = ProjectData.serialization.fromJson(
-          await editorStore.depotServerClient.getProject(
-            editorStore.query.groupId,
-            editorStore.query.artifactId,
-          ),
-        );
-        const currentQueryInfo =
-          await editorStore.graphManagerState.graphManager.getQueryInfo(
-            editorStore.query.id,
-          );
         if (
           registrationState.isInProgress ||
           !servicePattern ||
           !isServiceUrlPatternValid ||
-          !selectedEnvOption
+          !selectedEnvOption ||
+          !queryBuilderState.mapping ||
+          !(queryBuilderState.runtimeValue instanceof RuntimePointer)
         ) {
           return;
         }
         try {
           registrationState.inProgress();
-          const serverUrl = guaranteeNonNullable(
-            serverRegistrationOptions.find(
-              (option) => option.env === selectedEnvOption.value,
-            )?.managementUrl,
+          const serverConfig = serverRegistrationOptions.find(
+            (option) => option.env === selectedEnvOption.value,
           );
-          const versionInput = LATEST_PROJECT_REVISION;
+          const queryContent =
```

Comment:
Have we looked at what `createServiceElement` does?
We are making a `lambdaToPureCode` and calling `pureCodeToLambda` for the lambda again.